kernel/pmu_x86/src/lib.rs

```diff
@@ -701,7 +704,7 @@ pub fn start_samples(event_type: EventType, event_per_sample: u32, task_id: Opti
     // This check can never trigger since `event_per_sample` is a `u32`
     // and is therefore by definition in the range `u32::MIN..=u32::MAX`.
     // We'll check anyways, just in case `event_per_sample`'s type is changed.
-    if u32::try_from(event_per_sample).is_ok() {
+    if event_per_sample.type_id() == TypeId::of::<u32>() {
```

Comment:
TypeIDs aren't generally supposed to be used for this purpose. Let's keep it the way it was, since we do want a u32 out of it, and simply add an `allow` attribute above it.

Reply:
Why aren't TypeID:s supposed to be used for this purpose? The way it was triggered a lint (I don't remember which one), and the way I have it now doesn't trigger any lints

Reply:
the purpose of this condition is to check whether `event_per_sample` remained in the type `u32`, or throw error if the type has changed to something other than `u32`

Reply:
Right,  but that's more of a concession for the fact that it needs to be changed anyway (see #539), so this is a temp fix. More specifically, the proper condition is not that it _must_ be a `u32`, but that the value of `event_per_sample` should fit into a `u32`. The old code is more correct.